### ISSUE TITLE
Port forwarding for WSL doesn't work on other devices

### ISSUE THREAD

**Issue by MyComputerIsBig** [oobabooga/text-generation-webui#380]
When I start the webui on Windows with the --listen argument, I can connect my smartphone to it, however it doesn't work when I start the webui on WSL.

My computer ipv4 is 192.168.0.31, my smartphone is 192.168.0.37. They are both connected to my box.

I followed [these instructions](https://github.com/oobabooga/text-generation-webui/wiki/Windows-Subsystem-for-Linux-(Ubuntu)-Installation-Guide), especially this command:
`netsh interface portproxy add v4tov4 listenaddress=0.0.0.0 listenport=7860 connectaddress=localhost connectport=7860`

Thanks to this command I can now connect to the webui on my computer (192.168.0.31) with the following urls (both of them work): `http://127.0.0.1:7860` and `http://192.168.0.31:7860`

But none of them work on my smartphone (192.168.0.37).
Did I miss something?

**Comment by sgsdxzy** [comment]
I think you also need to setup Windows firewall to allow inbound tcp traffic to that port:
https://www.softwaretestinghelp.com/how-to-open-ports/#:~:text=Go%20to%20the%20Advanced%20settings,connection%20and%20click%20on%20Next.

**Comment by MyComputerIsBig** [comment]
Well actually it worked for 10 minutes but when I restarted the webui it completely stopped worked, even on my computer...
I tried to restart my computer, I also tried to close WSL with `wsl --shutdown` but none of these solution worked.

**Comment by sgsdxzy** [comment]
I find out you need to remove the port forwarding first:
`netsh interface portproxy reset`
Then start webui, then re-add the port forward.
Or if you want something permanent, launch webui with --listen and use wsl's internal ip, and forward to that ip in netsh also.

**Comment by MyComputerIsBig** [comment]
Thank you, your first solution worked very well, but I am not a fan of it. I would prefer your second solution but I am not sure I am doing it right:

I did `wsl hostname -i` to get the ip adress of wsl (127.0.1.1)
Then I did `netsh interface portproxy reset`
Followed by `netsh interface portproxy add v4tov4 listenaddress=0.0.0.0 listenport=7860 connectaddress=127.0.1.1 connectport=7860`

Now I can connect to the webui from my computer but it doesn't work on other devices. I have the --listen option enabled.

**Comment by sgsdxzy** [comment]
> Thank you, your first solution worked very well, but I am not a fan of it. I would prefer your second solution but I am not sure I am doing it right:
> 
> I did `wsl hostname -i` to get the ip adress of wsl (127.0.1.1), Then I did `netsh interface portproxy reset`, Then I started the server, Followed by `netsh interface portproxy add v4tov4 listenaddress=0.0.0.0 listenport=7860 connectaddress=127.0.1.1 connectport=7860`
> 
> Now I can connect to the webui from my computer but it doesn't work on other devices. I have the --listen option enabled. By the way, the problem remains, I have to reset the port forwarding every time before I start the server

You need the "internal" ip of the wsl guest. Type `ifconfig` in wsl, it should be something like 172.20.36.1

**Comment by MyComputerIsBig** [comment]
Thank you !

**Comment by 9cento** [comment]
TL;DR: I did the procedure, it worked, I un-did it because stuff, I did it again (and again, double-checking it) but it doesn't work anymore no matter what (can't connect from another device on the same private network). What's happening?

So I did everything posted here and managed to get the thing to work as intended, I also put in place a script in wsl ubuntu to revert its ip to the one specified in `netsh interface portproxy add...` (so pseudo-static) and the setup was even surviving reboots. Then for an unrelated issue (wsl ubuntu wasn't able to connect to the internet anymore, will investigate later) I did `netsh interface portproxy reset`, proceeded to do my stuff with the re-enabled internet connection and then re-did the procedure from scratch again (including `netsh interface portproxy reset`), to my surprise the procedure didn't work anymore. How comes?
